/*
 *    Copyright 2020 Google LLC
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        https://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */
package simproto;

message Command {
        required string cwd = 1;
        required string command = 2;
        repeated string args = 3;
}

message ApproveRequest {
        optional string host = 1;
        optional string user = 2;
        optional Command command = 3;
	optional string justification = 4;
}

message ApproveResponse {
        required bool approved = 1;
	optional string comment = 2;
}

message SimConfig {
        // Where sockets are created.
        required string sock_dir = 1;

        // Administrator group.
        optional string admin_group = 2;

        // Group owner of the socket. AKA approvers group.
        optional string approve_group = 3;
}
